File: 프로그래머스/3/12907. 거스름돈/거스름돈.java
class Solution {
    long nrm = 1_000_000_007;
    
    public int solution(int n, int[] money) {
        long dp[] = new long[n+1];
        for (int i=0; i<=n; i++) dp[i] = 0;
        
        for (int coin : money) {
            dp[coin] += 1;
            for (int i=coin; i<=n; i++) {
                dp[i] =  (dp[i] + dp[i-coin]) % nrm; 
            }
        }
        
        return (int) dp[n];
    }
}

File: 프로그래머스/3/12907. 거스름돈/거스름돈.py
def solution(n, money):

    dp = [0 for _ in range(n+1)]
    dp[0] = 1 # 초기 조건 

    for num in money: # O(100)
        for amount in range(num, n+1): # O(100000)
            dp[amount] += dp[amount-num]
    # O(100 x 100000)
    return dp[n]